src/v/datalake/record_multiplexer.cc

```diff
@@ -49,13 +51,17 @@ record_multiplexer::operator()(model::record_batch batch) {
         iceberg::struct_value data = translator.translate_event(
           std::move(key), std::move(val), timestamp, offset);
         // Send it to the writer
-        auto& writer = get_writer();
-        data_writer_error writer_status = co_await writer.add_data_struct(
-          std::move(data), estimated_size);
-        if (writer_status != data_writer_error::ok) {
+
+        try {
+            auto& writer = co_await get_writer();
+            _writer_status = co_await writer.add_data_struct(
+              std::move(data), estimated_size);
+        } catch (const std::runtime_error& err) {
+            datalake_log.error("Failed to add data to writer");
```

Comment:
`vlog`

Also we should set the `_writer_status` here.

Reply:
nit: I think _writer_status should probably be initialized to a generic_error and be set to ok when it actually finished.